app/src/main/java/ch/epfl/skysync/screens/FlightCalendar.kt

```diff
@@ -0,0 +1,194 @@
+ *
+ * @param date The date for which the flight information is requested.
+ * @param time The time slot AM or PM.
+ * @param getFlightByDate A function that retrieves flight information based on date and time. It
+ *   takes a LocalDate and a TimeSlot as input parameters and returns a Flight object representing
+ *   the flight information for the specified date and time slot, or null if no flight exists for
+ *   that date and time slot.
+ * @param onClick Lambda function representing the action to perform when the button is clicked.
+ */
+@Composable
+fun ShowFlight(date: LocalDate, time: TimeSlot, viewModel: UserViewModel) {
+  val flight = viewModel.user.value.assignedFlights.getFirstFlightByDate(date, time)
+  var weight = 0.5f
+  if (time == TimeSlot.PM) {
+    weight = 1f
+  }
+  if (flight != null) {
+    Button(
+        onClick = {},
+        shape = RectangleShape,
+        modifier = Modifier.fillMaxHeight().fillMaxWidth(weight),
+        colors = ButtonDefaults.buttonColors(containerColor = Color.Cyan)) {
+          Text(
+              text = flight.flightType.name,
+              fontSize = 12.sp,
+              color = Color.Black,
+              overflow = TextOverflow.Clip,
+              maxLines = 1,
+              modifier = Modifier.width(120.dp),
+              textAlign = TextAlign.Center)
+        }
+  } else {
+    Box(
+        modifier = Modifier.fillMaxHeight().fillMaxWidth(weight),
+        contentAlignment = Alignment.Center) {}
+  }
+}
+/**
+ * Composable function to display the flight calendar screen.
+ *
+ * @param navController The navigation controller used for navigating to different destinations
+ *   within the app.
+ */
+@Composable
+fun ShowFlightCalendar(navController: NavHostController, viewModel: UserViewModel) {
+  Scaffold(bottomBar = { BottomBar(navController) }) { padding ->
+    Box() { Calendar(navController, viewModel, padding) }
+  }
+}
+
+/**
+ * Composable function to display a calendar with flight information for each date and time slot.
+ *
+ * @param navController The navigation controller used for navigating to different destinations
+ *   within the app.
+ */
+@Composable
+fun Calendar(navController: NavHostController, viewModel: UserViewModel, padding: PaddingValues) {
+  var currentWeekStartDate by remember { mutableStateOf(getStartOfWeek(LocalDate.now())) }
+  Column(
+      modifier = Modifier.fillMaxSize().padding(16.dp).background(Color.White),
+  ) {
+    WeekView(currentWeekStartDate, viewModel, navController)
+    Spacer(modifier = Modifier.height(8.dp))
+    Row(
+        modifier = Modifier.fillMaxWidth().padding(vertical = 10.dp),
+        horizontalArrangement = Arrangement.SpaceBetween) {
+          Spacer(modifier = Modifier.width(5.dp))
+          Button(
+              onClick = { currentWeekStartDate = currentWeekStartDate.minusWeeks(1) },
+              colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFFFA500))) {
```

Comment:
For raw values like code colors, it is strongly advised to use global variables that you initialize at the beginning of the code for better modularity :)